auth/standard_role_manager.cc

```diff
@@ -278,6 +277,10 @@ future<> standard_role_manager::stop() {
     return _stopped.handle_exception_type([] (const sleep_aborted&) { }).handle_exception_type([](const abort_requested_exception&) {});;
 }
 
+future<> standard_role_manager::ensure_superuser_is_created() {
+    return _superuser_created.get_future();
```

Comment:
I wonder is it safe to call it several times? I see there is some assert `assert(!this->_future && this->_state && !this->_task);` 

Asking because `ensure_superuser_is_created` doesn't limit how many times it can be called.

Reply:
> I wonder is it safe to call it several times? 

No, it's not :) To make it safe one need to use shared_future-s or play games with some other indicator member (bool called_once or make _superuser_created a pointer)

Reply:
maybe semaphore would be enough instead of future+bool/flag ?

Reply:
seastar::semaphore::wait wouldn't do the job?

Reply:
you set it to 0, increment by 1 after superuser created, and wait in ensure_superuser_is_created. it would also serialize ensure_superuser_is_created calls but that's not a big problem